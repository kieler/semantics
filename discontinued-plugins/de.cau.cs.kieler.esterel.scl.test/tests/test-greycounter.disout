module Counter:
input T;
input S;
input S0;
input S1;
input S2;
input S3;
output B0;
output B1;
output B2;
output B3;
output ALARM;

signal R,
       T5,
       C0,
       C1,
       C2,
       A0,
       A1,
       A2 in
    every 5 T do
      emit T5
    end every
  ||
    loop
      abort
        loop
          abort
            loop
              pause
            end loop
          when T5;
          abort
            loop
              emit B0;
              pause
            end loop
          when T5;
          emit C0;
          abort
            loop
              emit B0;
              pause
            end loop
          when T5;
          abort
            loop
              pause
            end loop
          when T5;
          emit C0
        end loop;
        loop
          pause
        end loop
      when R
    end loop
  ||
    loop
      abort
        loop
          abort
            loop
              pause
            end loop
          when C0;
          abort
            loop
              emit B1;
              pause
            end loop
          when C0;
          emit C1;
          abort
            loop
              emit B1;
              pause
            end loop
          when C0;
          abort
            loop
              pause
            end loop
          when C0;
          emit C1
        end loop;
        loop
          pause
        end loop
      when R
    end loop
  ||
    loop
      abort
        loop
          abort
            loop
              pause
            end loop
          when C1;
          abort
            loop
              emit B2;
              pause
            end loop
          when C1;
          emit C2;
          abort
            loop
              emit B2;
              pause
            end loop
          when C1;
          abort
            loop
              pause
            end loop
          when C1;
          emit C2
        end loop;
        loop
          pause
        end loop
      when R
    end loop
  ||
    signal SMEM,
           SMEM1 in
        loop
          if (SMEM) {
            pause;
            emit SMEM1
          } else {
            pause
          }
        end loop
      ||
        loop
          if (S) {
            if (S0) {
              emit SMEM
            }
          } else {
            if (SMEM1) {
              emit SMEM
            }
          };
          if (tick and (SMEM and B0 or not SMEM and not B0)) {
            emit A0
          };
          pause
        end loop
    end signal
  ||
    signal SMEM,
           SMEM1 in
        loop
          if (SMEM) {
            pause;
            emit SMEM1
          } else {
            pause
          }
        end loop
      ||
        loop
          if (S) {
            if (S1) {
              emit SMEM
            }
          } else {
            if (SMEM1) {
              emit SMEM
            }
          };
          if (A0 and (SMEM and B1 or not SMEM and not B1)) {
            emit A1
          };
          pause
        end loop
    end signal
  ||
    signal SMEM,
           SMEM1 in
        loop
          if (SMEM) {
            pause;
            emit SMEM1
          } else {
            pause
          }
        end loop
      ||
        loop
          if (S) {
            if (S2) {
              emit SMEM
            }
          } else {
            if (SMEM1) {
              emit SMEM
            }
          };
          if (A1 and (SMEM and B2 or not SMEM and not B2)) {
            emit A2
          };
          pause
        end loop
    end signal
end signal

end module
