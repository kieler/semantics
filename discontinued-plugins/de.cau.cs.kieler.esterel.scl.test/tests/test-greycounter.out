module Counter:
input T;
input S;
input S0;
input S1;
input S2;
input S3;
output B0;
output B1;
output B2;
output B3;
output ALARM;

signal R,
       T5,
       C0,
       C1,
       C2,
       A0,
       A1,
       A2 in
    every 5 T do
      emit T5
    end every
  ||
    run Bit [
      signal T5 / CLK,
             B0 / B,
             C0 / CY ]
  ||
    run Bit [
      signal C0 / CLK,
             B1 / B,
             C1 / CY ]
  ||
    run Bit [
      signal C1 / CLK,
             B2 / B,
             C2 / CY ]
  ||
    run BitAlarm [
      signal B0 / B,
             S0 / SX,
             tick / ACI,
             A0 / ACO ]
  ||
    run BitAlarm [
      signal B1 / B,
             S1 / SX,
             A0 / ACI,
             A1 / ACO ]
  ||
    run BitAlarm [
      signal B2 / B,
             S2 / SX,
             A1 / ACI,
             A2 / ACO ]
end signal

end module

module Bit:
input R;
input CLK;
output B;
output CY;

loop
  loop
    await CLK;
    abort
      sustain B
    when CLK;
    emit CY;
    abort
      sustain B
    when CLK;
    await CLK;
    emit CY
  end loop
each R

end module

module BitAlarm:
input B;
input S;
input SX;
input ACI;
output ACO;

signal SMEM,
       SMEM1 in
    loop
      present SMEM then
        pause;
        emit SMEM1
      else
        pause
      end present
    end loop
  ||
    loop
      present S then
        present SX then
          emit SMEM
        end present
      else
        present SMEM1 then
          emit SMEM
        end present
      end present;
      present [ACI and (SMEM and B or not SMEM and not B)] then
        emit ACO
      end present;
      pause
    end loop
end signal

end module
